src/linear_algebra/tensor_product.lean

```diff
@@ -115,39 +116,39 @@ variables {R M}
 
 end linear_map
 
-variables {R : Type*} [comm_ring R]
+section semiring
+variables {R : Type*} [comm_semiring R]
 variables {M : Type*} {N : Type*} {P : Type*} {Q : Type*} {S : Type*}
 
-variables [add_comm_group M] [add_comm_group N] [add_comm_group P] [add_comm_group Q] [add_comm_group S]
-variables [module R M] [module R N] [module R P] [module R Q] [module R S]
+variables [add_comm_monoid M] [add_comm_monoid N] [add_comm_monoid P] [add_comm_monoid Q] [add_comm_monoid S]
+variables [semimodule R M] [semimodule R N] [semimodule R P] [semimodule R Q] [semimodule R S]
 include R
 
 variables (M N)
 
 namespace tensor_product
 
 section
-open free_abelian_group
+-- open free_add_monoid
```

Comment:
Do you want to keep this comment?

Reply:
Yes. See https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/cannot.20open.20namespace.20free_add_monoid

Reply:
okido